### ISSUE TITLE
[BUG] Weird disconnection every once in a while only on a specific PC

### ISSUE THREAD

**Issue by Morgawr** [streetpea/chiaki-ng#449]
**Describe the bug**
On my Linux PC, I play with chiaki-ng connected to my PS5, however I only use chiaki as a "monitor", as I have my PS5 next to me and use the controller directly connected to it (I cannot set up the PS5 controller to work well via chiaki on linux so I prefer to use it straight up to the PS5).

To do this, I log into the PS5 from chiaki-ng using my secondary account, and then I swap to my primary account via the controller.  (not sure if this is relevant).

Every once in a while, during a play session, chiaki just straight up disconnects and I have to re-connect to continue playing. This seems very inconsistent as sometimes it happens multiple times in a short timespan (like maybe 10 or so times in 30 minutes) and some other times it never happens (I just had a 3-hour long playing session with 0 disconnects). 

I also play on this PS5 using my steam deck with chiaki-ng (but without the double account setup), same network connection and everything, and it never happens. This only happens when I play with the dual account set up on Linux.

**Debug Log**

```
[I] Remote disconnected from StreamConnection with reason "Server shutting down"
[I] StreamConnection is disconnecting
[I] StreamConnection sending Disconnect
[I] StreamConnection closing after Remote disconnected
[I] StreamConnection closed takion
[E] Remote disconnected from StreamConnection
[I] Ctrl requested to stop
[I] Ctrl stopped
[I] Session has quit
[I] Discovery failed to bind port 9304, trying one higher
[I] chiaki_holepunch_session_fini: Holepunch session deleted.
[E] websocket_thread_func: Select canceled.
```

**To Reproduce**
1. Start chiaki-ng and connect to a PS5 with a secondary account
2. connect a controller directly to the PS5 and swap to the primary account
3. play 
4. eventually it disconnects

**Expected behavior**
To not disconnect

**Desktop (please complete the following information):**
 - OS: Arch Linux x86_64
 - Kernel: Linux 6.11.2-arch1-1
 - DE: KDE Plasma 6.2.2
 - Device: PC (Intel(R) Core(TM) i7-8700K (12) @ 4.70 GHz, NVIDIA TITAN X, 64GB ram)
 - chiaki-ng Version 1.9.0 (from flatpak)

**Comment by streetpea** [comment]
@Morgawr it looks like you are using remote connection via psn. Is there a reason you aren’t using a local connection? Anyway, it looks like like you’re ps5 is shutting down or going into rest mode and that is causing the stream to end.

**Comment by Morgawr** [comment]
Ah yes sorry I forgot to mention it, my PS5 is on a different network from my PC. They are two sub-LANs from the same home connection. It's because I work from home and have two networks separated but the PS5 has to be on the other network (because it's wired to a different router). This shouldn't be an issue though as it works the same way on my steam deck which does not present this behaviour.

> Anyway, it looks like like you’re ps5 is shutting down or going into rest mode and that is causing the stream to end.

It's not going into rest mode, that's the problem. I have also a capture card connected to my PC so I can see the screen in real-time (also I can hear the audio from my wireless headphones), it just says "remote play has disconnected". And for the reason why I use remote play is because I get better quality/framerate using it than my capture card, funnily enough. 

> Do you have hdmi cec enabled and something connected to your ps5 via hdmi? If that’s the case it could be getting signals to turn from a connected tv for instance.

I mentioned my capture card, but most of the time when I play it's not connected to anything at all (I only connect it to the capture card to debug stuff in case the connection drops, like in this situation, but it happens even without it). Other than that, it's not connected to anything.

**Comment by streetpea** [comment]
@Morgawr can you post a full log with debug logging enabled? In the logs you posted it says: “Remote disconnected from StreamConnection with reason "Server shutting down" this will only happen if the console sends a protobuf message with the quit reason of “server shutting down”. Chiaki-ng is then terminating the session because it is being told to by the PlayStation because it says it’s shutting down (going into sleep mode or turning off). If this log is from a session that exhibits the behavior you described then it seems the ps is initiating the disconnection.

**Comment by Morgawr** [comment]
Yeah I can 100% confirm the PS is not shutting down/going to sleep. My suspicion is that *maybe* since I am playing on a secondary (local) account and the "main" account connected via remote is "idle", the PS5 decides to stop the remote connection. However what I cannot explain is why it is so inconsistent, sometimes I can go for hours just fine and sometimes it disconnects every few minutes. I even tried making sure to hit a random button in chiaki (remote connection) just to signal that it's still active but it still happens anyway.

As for the debug logs, how do I do that? I am running in flatpak with the command:

```shell
$ /usr/bin/flatpak run --branch=stable --arch=x86_64 --command=chiaki io.github.streetpea.Chiaki4deck
```

**Comment by streetpea** [comment]
@Morgawr there is a checkbox on the first tab of the settings for verbose logging.

**Comment by Morgawr** [comment]
I have attached my chiaki session verbose logs, I scrubbed them of any identifiable information/strings (let me know if I missed something).

This is a full session of about 12 minutes until the problem presented itself. 

[chiaki_session_2024-10-29_13-43-33-372372.log](https://github.com/user-attachments/files/17551848/chiaki_session_2024-10-29_13-43-33-372372.log)

**Comment by streetpea** [comment]
@Morgawr I did notice a small error. Not sure if it has anything to do with your issue though. Anyway here is a new build with that change: https://github.com/streetpea/chiaki-ng/actions/runs/11699190530/artifacts/2150830593
